Let assembly reloading bypass undo/redo to improve editor stability

adrsch opened the pull request:
# PR Details

Hack to prevent serious editor bugs on undo/redo for assembly reloads, while leaving the code for it all intact. Introduces poor UI behavior on the Edit History tab, but that's never caused anyone to lose work.

## Description

Add a transaction flag for bypassing the undo/redo system, which should not be used except for hacks.

Tested by moving things around and/or editing properties, doing stuff to cause an assembly reload, and then using undo. The things done before the editor reload, successfully undo.

## Related Issue

<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
https://github.com/stride3d/stride/issues/2107

## Motivation and Context

Pain

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] **I have built and run the editor to try this change out.**

adrsch committed:
Let assembly reloading bypass undo/redo to improve editor stability

adrsch commented:
@dotnet-policy-service agree

Kryptos-FR commented:
That's not the way to do it. There is already infrastructure in place to have not-cancellable operations.

With that said, I seem to remember (old story from when I was at SiliconStudio) that there was a reason we still needed that operation to be cancellable. I think it was related to serialization.

Kryptos-FR commented:
In also added a comment in the related issue. At this stage, unless it can be reproduced and confirmed, a hack should not be merged without measuring its consequences.

adrsch commented:
> That's not the way to do it. There is already infrastructure in place to have not-cancellable operations.
> 
> With that said, I seem to remember (old story from when I was at SiliconStudio) that there was a reason we still needed that operation to be cancellable. I think it was related to serialization.

I figured this hack was terrible! 

Hopefully the standard reproduction I wrote up on the issue a few minutes ago, using a sample project, can help it get fixed properly, I mostly just figured it would be good to get a hack out there. Since this seems like a thing some people will run into not at all, or all the time, depending on their habits with using undo in the editor.

adrsch commented:
I got the true cause figured out of duplicate scripts getting added in https://github.com/stride3d/stride/issues/2107#issuecomment-1892788016 

Not sure how to fix, it seems like a problem that has got to have been addressed elsewhere in the engine (an item getting processed twice, once as a member of a scene, and second as a member of a prefab)

Eideren committed:
Strip forced bepu rebuild on opening up the demo, too time-consuming

Eideren committed:
Use cc0 assets

Doprez committed:
add math conversions

Doprez committed:
Add basic navigation agent

Doprez committed:
Add physics navigation agent

Doprez committed:
Merge branch 'eideren-bepu_integration' into doprez-bepu-integration

# Conflicts:
#	build/Stride.sln
#	sources/engine/Stride.BepuPhysics/Stride.BepuPhysics.Navigation/Definitions/BuildSettings.cs
#	sources/engine/Stride.BepuPhysics/Stride.BepuPhysics.Navigation/Processors/RecastMeshProcessor.cs
#	sources/engine/Stride.BepuPhysics/Stride.BepuPhysics.Navigation/StrideGeomProvider.cs

Eideren committed:
Merge remote-tracking branch 'refs/remotes/Doprez/master' into bepu_integration

Eideren committed:
Merge Doprez Navigation Changes

Eideren committed:
Use dotnet runtime's editorconfig instead of inheriting from Stride's - the project started out as a stand-alone solution outside this repo, we should silo its code style but still enforce one at the very least

Eideren committed:
Apply coding style over new recast changes

Eideren committed:
Fix null issues, minor style tweaks and warnings

Eideren committed:
Merge remote-tracking branch 'refs/remotes/Doprez/master' into gizmo

# Conflicts:
#	sources/editor/Stride.Assets.Presentation/StrideDefaultAssetsPlugin.cs

Eideren committed:
Handle newly introduced ModelGizmo post merge

Eideren committed:
Merge branch 'refs/heads/gizmo' into bepu_integration

Eideren committed:
Specify that changes to colliders may reset internal physics state

Eideren committed:
Remove CharacterComponent's debug info

Eideren committed:
Minor changes and more annotate one more callsite to check with Norbo

Eideren committed:
Rewrite constraints backend, introduce Constraint list for BodyComponent

Eideren committed:
Create Test project covering BepuPhysics, starting with constraints

Eideren committed:
Fix ListOfColliders workaround, in turn fixing some issues with gizmos

Eideren committed:
Basic Constraint Gizmos

Eideren committed:
Enable/Disable gizmos

Eideren committed:
Remove deprecated Stride.Physics reference

Eideren committed:
Rename Stride.BepuPhysics.DebugRender to Debug

Eideren committed:
Update .sdscene based on naming and structure changes

Eideren committed:
Rollback to EntityProcessor

Eideren committed:
Improve collider change tracking

Eideren committed:
Ensure adding the debug component before adding collidables still sets the visibility appropriately

Doprez committed:
Make FastList Obsolete (#2332)

Doprez committed:
Fix a couple of .NET 6 reference we missed (#2330)

* .NET 6 reference missed

* added central version number from Jkao for better consistancy

* updated comment

* Get version from Environment

azeno committed:
Fixes assembly load failure when building Stride based project on clean machine (2. attempt, issue #2232) (#2319)

Basewq committed:
[Graphics] Remove signed blend indices due to dropping OpenGL ES 2 support, and fix model skinning on platforms using OpenGL ES (ie. Android and iOS) (#2323)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

YerkoAndrei committed:
[GameStudio] Prevents a null IMemberNode to update (#2307)

* prevents a null IMemberNode to update

* checking properties before calling update

Jklawreszuk committed:
feat: Adjust freeimage to all platforms (#2303)

* feat: Adjust freeimage to all platforms

* Adjust FreeImage methods

Doprez committed:
fix: AssImp orientation, imported material color fallback, and set default animation preview model (#2291)

* ported https://github.com/assimp/assimp/pull/5494/files

* Cleanup

* Potential fix to investigate further

* Fallback to BaseColor when diffuse is not available

* Automatically set preview model on import for animations

* Use BaseColor as Diffuse if Diffuse is not defined

* Document pivot todo

* Rollback transformations related logic

* More cleanup

* Fix issue when trying to import animation from an fbx without animations

---------

Co-authored-by: Eideren <contact@eideren.com>

Eideren committed:
[Editor] Disable ModelComponent Gizmos by default (#2321)

Jklawreszuk committed:
feat: Adjust DirectTex for all platforms (#2314)

Eideren committed:
feat: FlexibleProcessing as alternative to EntityComponentProcessor for lightweight, abstract and typesafe processing in idiomatic c# (#2331)

* feat: FlexibleProcessing as alternative to EntityComponentProcessor for lightweight, abstract and typesafe processing in idiomatic c#

* is ... == false to is not

* Access backing buffer of the list to improve get/set operations

* More tests for edge cases

Eideren committed:
Revert "feat: Adjust DirectTex for all platforms (#2314)" (#2337)

This reverts commit f23ce73fbb406875f9dabce553e326a55aec12a6.

Jklawreszuk committed:
feat: Adjust DirectTex to all platforms 2 (#2338)

Basewq committed:
[Editor] Fix displaying & editing project User Settings (#2350)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

Jklawreszuk committed:
fix: Update Stride.Core.Assets.CompilerApp.targets file (#2360)

Jklawreszuk committed:
fix: Enable to run broken TextureConverter Tests (#2343)

tymokvo committed:
Update guillotine packer documentation link (#2339)

* Use Wikipedia link in class documentation

* Use PDF from author github

Arc-huangjingtong committed:
Fix Compile Warning in OpenXR (#2357)

* Delete the useless namespace

* Revert "Delete the useless namespace"

This reverts commit 454df6c0db279b5d82869880d1d24c76f06d7185.

* Delete the useless namespace

Basewq committed:
[Core] Fix getting application executable path when using PublishSingleFile (#2352)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

Basewq committed:
fix importing effects to the existing EffectCompileLog (#2362)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

Basewq committed:
[Font] Update SharpFont library for additional fixes (#2318)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

Arc-huangjingtong committed:
Update : geometric primitive.cylinder to avoid GC (#2369)

* Update GeometricPrimitive.Torus.cs

* Update GeometricPrimitive.Torus.cs

* Update GeometricPrimitive.Torus.cs

* Update GeometricPrimitive.Cylinder.cs

* Revert "Update GeometricPrimitive.Torus.cs"

This reverts commit 2db32fde9ae8ba9e610b1ab0cc23c37c5481cb26.

* Reapply "Update GeometricPrimitive.Torus.cs"

This reverts commit e8bbe833197c440808984ddb9ef34cf9bd03382a.

Arc-huangjingtong committed:
Update : GeometricPrimitive.Torus.cs (#2368)

* Update GeometricPrimitive.Torus.cs

* Update GeometricPrimitive.Torus.cs

* Update GeometricPrimitive.Torus.cs

Jklawreszuk committed:
fix: [DirectoryWatcher] - Folders are case sensitive on Linux (#2363)

* fix: [DirectoryWatcher] - Folders are case sensitive on Linux

* Add TODO comments

minktusk committed:
Update launcher brand icons (#2345)

* Update launcher brand icons

* Update launcher brand icons

* More quality images

Jklawreszuk committed:
feat: Modernize PVRTexLib dependency (#2342)

* Modernize PVRTexTool dependency

* Update outdated info

timcassell committed:
Fixed a bug with pooling ChannelMicroThreadAwaiter. (#2373)

Jklawreszuk committed:
fix: Replace ToWindowsPath with ToOSPath (#2371)

tebjan committed:
Fixes depth calculation for MipMap views of 3D textures (#2375)

Co-authored-by: tebjan <tebjan@gmail.com>

Eideren committed:
Surface index of child hit by physics tests

Jklawreszuk committed:
fix: Build folder cleanup (#2383)

Basewq committed:
[Font] Fix glyph bitmap alignment from SDF fonts, dump generated SDF bitmaps in debug build into its own folder in the temp directory instead of straight into the temp directory (#2380)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

Jklawreszuk committed:
feat: Hide default font name for SystemFontProvider (#2385)

Jklawreszuk committed:
fix: Typo in VRSandboxApp (#2396)

dloe committed:
feat: Save Asset View Options Settings (#147) (#2320)

* Zoom save settings for Tile and Grid

* added ViewFilters Settings Key

* Adjusted settingskey and added load area

* Fixed formatting on SettingsKey default

* Implemented Filters to be saved and reloaded on new instances

* Added SaveAssetFilters call to filter IsActive toggle

* Adjusted comments and fixed filter load order to be consistent

* Updated load and save methods to now be private

- No reason for them to be public

* Removed unnecessary comment

- was for tracking potential load points with asset  filters

* added private access modifier for asset filter view model data var

---------

Co-authored-by: soorMSWE <soor@uci.edu>
Co-authored-by: Will S <wrshield@uci.edu>

Eideren committed:
Fix collidable starting empty failing to reattach

Kryptos-FR committed:
fix: Bump samples version (#2398)

Needed after #2109

Eideren committed:
feat: Expose and cleanup editor gizmos (#2127)

Jklawreszuk committed:
fix: Remove unused data (#2389)

Jklawreszuk committed:
fix: Change the error to a warning when searching for font path (#2403)

Basewq committed:
Fix shader bool generics to appear in the editor (#2405)

* [Rendering] Fix shader generics with bool type to appear in the editor

* [Rendering] Simplify type checking ComputeColorParameter types

---------

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

Eideren committed:
Fix interpolation being ignored unless specifically toggled while in a scene

Basewq committed:
fix: Add back Windows specific CompilerApp to fix libraries with WPF references (#2392)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

dloe committed:
feat: No Physics Shapes On The CharacterComponent Causes Unhandled Exception  (#2410)

* Added early warning and better error logging for CharacterComponent missing physical shapes

- Will now incorporate logger with a warning when Attachments of shapes for KinecticCharacter are skipped due to no ColliderShapes being present.
- errors given inside CharacterComponent will also include call info for users to trace and find what corresponding code is calling the physic method.

* Cleaned up comments

+ fixed typo in comment

* Added extra check for No Shapes warning

Found reference to older code that may have at one point had this check and warning. Added the extra check I had not accounted for

* Removed repetitive check for lack of collider shap

- Removed check that occurs after DisableSimulation check and now has it ahead of simulation check

* Moved comment to summary of PhysicsComponent OnAttach

Cosmetic change with comments and summary

* Reduced code reuse with dedicated function for logging

Maybe it could go ever farther and be put in the Logger class? May update

Jklawreszuk committed:
Fix sample projects, remove unused properties (#2205)

dloe committed:
fix: PhysicsComponent created with GetOrCreate<T> is unusable (#2422)

* Run proper Rigidbody setup when adding ColliderShape

* Added test case for proper setup with GetOrCreate

* Followed similar Bepu format with creation of ReAttach method

* Reformatted files edited

* Revert "Reformatted files edited"

This reverts commit 1c9b03eb6e60fcb18a6fe8d55f0df715119ad5e7.

VaclavElias committed:
fix: Xml comments fix 5 (#2419)

* fix: Correct reference added to BoundingBox

* fix: cref corrected - removed full stop

* fix: cref for AudioEmitterComponent corrected

* fix: param fixed from rotation to scale

* fix: RaiseException doesn't exist, most likely it should be SetException, which is present

* fix: cref corrected with full namespace

* fix: cref references corrected

* chore: Reversing the update because it's not working

* fix: Incorrect reference

* fix: The C# reference itself doesn't exist but the reference seems ok

* fix: Correcting the references

* fix: Fixed xml comment

* fix: xml summary corrected

* fix: cref reversed back, but added full namespace

* fix: Fixing see xml tag with appropriate html link to our manual

VaclavElias committed:
docs: Enhance Entity class comment with detailed description. (#2423)

The comment for the `Entity` class has been updated to provide a more detailed description. It now specifies that the class represents a game entity that typically aggregates multiple `EntityComponent` instances. Additionally, it includes references to the Stride documentation for more information about adding and managing entities.

Basewq committed:
[NuGet] Fix setting platform version on the target framework moniker (TFM) when passed to NuGet resolver to properly load NuGet libraries (#2432)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

Eideren committed:
Writing to either Body.Position and Body.Orientation set the transform appropriately

Eideren committed:
Virtualize and document CharacterComponent to allow users to customize their implementation

Eideren committed:
Merge remote-tracking branch 'STRIDE/master' into bepu_integration

# Conflicts:
#	sources/engine/Stride.Engine/Engine/Gizmos/IEntityGizmo.cs
#	sources/engine/Stride.Engine/Engine/OpaqueComponentId.cs

Eideren committed:
Fix naming scheme change post merge

Basewq committed:
fix: [TextureConverter] Restore LoadWICFile for Windows due to incomplete cross-platform functionality (#2435)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

dloe committed:
bug: Unable to Change CharacterComponent's collider shape properties without causing crash (#2428)

* somewhat messy Unoptimized Solution for UAF exception fix

* Multiple physics properties carry over on KinematicCharacter reConstruction

* Forgot to remove unneeded reference to NativeCollisionObject in reconstruction

* grammar issues with comments

VaclavElias committed:
docs: Added xml comment, and placed the namespace to correct position. Fixes #2306 (#2436)

Basewq committed:
[Importer] Fix extracting embedded texture files for materials, eg. from glTF file (#2441)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

Basewq committed:
[Importer] Fix Assimp skeleton/animation importer to link to the correct node when a model has duplicate node names (#2444)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

xen2 committed:
[Build] Switch to new .NET Foundation code signing tool

Eideren committed:
Rewrite of the collision mask feature

Eideren committed:
Improve editor UI for CollisionMask slightly

Basewq committed:
[Graphics] Fix SpriteBatch rotation logic to original code (rotate clockwise for increasing rotation value) (#2458)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

Eideren committed:
Merge remote-tracking branch 'STRIDE/master' into bepu_integration

Eideren committed:
fix: Cecil serialization inheritance cutoff with complex base types (#2457)

* fix: Cecil serialization inheritance cutoff with complex base types

* Add small summary of the loop

Eideren committed:
Fix unwritable collision mask

dloe committed:
fix: avoid Reattach to run before Entity properly attaches in setup (#2439)

NicusorN5 committed:
feat: [Math] - Exponential damping (#2461)

* Exponential damping

* ExpDecay double precision overload

* Flip arguments

Eideren committed:
Finish porting the last couple of invalid collision mask in the demo

Eideren committed:
FilterByDistance renamed to CollisionGroup, improve documentation

Eideren committed:
Default Value for CollisionGroup

Eideren committed:
Fix demo car collision mask

Basewq committed:
[Physics] Fix reattach logic where component hasn't been attached yet (#2465)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

Doprez committed:
fix: changed AssImp unsupported pivot bone error to warning (#2467)

Jklawreszuk committed:
feat: [Build] - Better Platform detection for Linux (#2460)

VaclavElias committed:
Update CODE_OF_CONDUCT.md

Jklawreszuk committed:
Remove unused Stride.Core.ConfigEditor (#2006)

Jklawreszuk committed:
[AssemblyProcessor] - Remove unused options and deadcode (#1979)

* Remove unused AutoNotifyProperty argument

* Remove not implemented Werror arg

* Remove unused signkeyfile option

* Remove unused targetFramework option

* Remove NotifyPropertyProcessor/PropertyChangedEventArgs

* MonoFixedProcessor is not gonna needed

* Undo removing TreatWarningsAsErrors props

* Fix argument spacing

IXLLEGACYIXL committed:
feat: Assembly Processor Module Initializer to Roslyn (#2402)

* make module initializer source generated

* disable roslyn

* fix order of module initializers : prio on assemblyscan and serialization

* fix shaders and update engine

* remove ModuleInitializerProcessor

* remove async bridge processor

* remove leftover module initializer processor

* remove unused init locals processor

* fix namespace for visual studio tools

---------

Co-authored-by: IXLLEGACYIXL <ixllegacy123@outlook.com>

Eideren committed:
chore: Build latest AssemblyProcessor changes (#2463)

Eideren committed:
feat: Custom Depth Test Function for Materials (#2415)

* [Graphics] Custom Depth Test Function for Materials

* Fix typo

IXLLEGACYIXL committed:
fix: Fix Cecil in master (#2477)

* Fix Cecil

* fix: remove unused addmoduleinitializer

---------

Co-authored-by: Eideren <contact@eideren.com>
Co-authored-by: IXLLEGACYIXL <ixllegacy123@outlook.com>

Eideren committed:
Merge remote-tracking branch 'STRIDE/master' into bepu_integration

Eideren committed:
Fix BepuConfiguration not being registered as a service by LoadBepuServices

Eideren committed:
Scene to simulation link

Eideren committed:
Nullable UrlReference for scenes, empty is not allowed by the serializer

Eideren committed:
Fix duplicate import warnings

Eideren committed:
Remove unnecessary nuget config

Eideren committed:
Remove unnecessary call to base implementation from empty virtual methods

Eideren committed:
Create one default simulation when there isn't any configuration setup

Eideren committed:
Remove demo projects from the main solution

Eideren committed:
awaitable simulation NextTick() and AfterTick()

Eideren committed:
Fix contact event on collidable removal

Eideren committed:
Add Contact Removal and Raycast removal tests

Eideren committed:
Add summary for "collisionMask" in ray tests + fix RaycastPenetrating naming scheme

Eideren committed:
Fix demo after naming scheme change

Eideren committed:
Align *Tick() naming scheme with ISimulationUpdate

Jklawreszuk committed:
feat: Make Stride.StorageTool as Gui Tool (#2479)

* feat: Make Stride.StorageTool as Gui Tool

* fixup! feat: Make Stride.StorageTool as Gui Tool

* fixup! feat: Make Stride.StorageTool as Gui Tool

* Update README.md

---------

Co-authored-by: Nicolas Musset <musset.nicolas@gmail.com>

kopffarben committed:
fix: Wrong Instanced call used for DrawInstanced in CommandList.Direct3D.cs (#2482)

CommandList.DrawInstanced(Buffer argumentsBuffer, int alignedByteOffsetForArgs = 0)
NativeDeviceContext was DrawIndexedInstancedIndirect but should be DrawInstancedIndirect

Eideren committed:
Fix manifold and improve character component handling

Eideren committed:
Character component: cut custom gravity, bepu will take care of it

Eideren committed:
Improve camera controls for demo character controller

Doprez committed:
Bepu Navigation fixes (#6)

* navigation component overhaul

* Removing redundant code

Doprez committed:
Bepu navigation clean up part 2 (#7)

* navigation component overhaul

* Removing redundant code

* move to GameSystem

* Updated DotRecast

* fix world transofrm move bug

* added build time property

* More settings!

* fix forward to be Z

* Eiderens a genius fixed move

* docs and extra options for tryfindpath

* omit string value

Jklawreszuk committed:
fix: Adjust Solution.cs for all platforms (#2472)

Jklawreszuk committed:
fix: Enable to build SkyBoxAsset using OpenGL (#2504)

Jklawreszuk committed:
fix: Resolve crashing issue for OpenGL UIPage (#2189) (#2500)

Eideren committed:
Revert "[Editor] Collection supports in abstract properties" as it is no longer necessary.

This reverts commit 3f710c690faa7600dc19daceff898200e1977e58.
This commit is part of another PR on Stride/master which will be merged at a later date.

Eideren committed:
Merge remote-tracking branch 'STRIDE/master' into bepu_integration

# Conflicts:
#	sources/Directory.Packages.props

Eideren committed:
Removed SimulationUpdateComponent in favor of ISimulationUpdate

Eideren committed:
Pass CollidableComponent instead of CollidableReference to IContactEventHandler - using CollidableReference in userland is unsafe

dloe committed:
Test case improvements and updates (#2393)

* TestHalf2 and TestHalf now can pass in suite

Half2 now has proper ToSting (before it would simply  output the type instead of the values with ToString). Converter seems to similarly match formats for other converts to a degree

* Added similar prints for Half3 and Half4, added comments on Converter

- Need verification on how the converters should be changed (if they even still need to be)
- All half tests now pass

* LifeTimeNoSimpleConstructor now passes

Test now passes but a bit confused as to what this  test is trying to do. Will need context with PR. Didnt have empty constructor therefore it was impossible for this to ever pass.

* RenderToWindow test now skips due to failure

Relates to a possible rework in GameTestBase and SendImage method

* Various Graphics Tests Using outdated images [DONT USE]

10 Windows Tests Updated
11 Windows Tests Updated

RunTestGame AMD test animation images are now uodated

RunTestGame seems to be using older images for animation tests

SpriteRender2DTests has outdated images

Updated SpriteRenderer2DTests.f1 and Tests files
(Test was failing for small pixel differences)

SpriteRenderer3DTests Updated Images

- Images were pixels off causing a failure

SpriteRotationTests updated images

- Images were pixels off causing a failure

SpriteTestGame updated images

- Images were pixels off causing a failure

TesselationTests updated images

- Images were pixels off causing a failure

SceneAmbientLight test has outdated images

Updated SceneAmbientLight image

* TEstGCHandleAlloc: Autolayout was set to fail when it does now in fact succeed

Unsure why this originally was set to false after reading documentation, might be missing some context. Can simply add a TODO above line 37 instead to give more context. The rest of the test seems to function as expected.

* Stride Particles Tests Updated Visual Tests

- Outdated images for various tests inside Particles.Tests.Windows

Stride.Graphics.Tests - Updated image for TestRenderToTexture Test

- Outdated images for AMD

* Updated TestGetFontInfo to now properly check if various Font Info is properly loaded

* Added skips for graphics related tests to avoid confusion and give additional info for why they fail info

* Updated template package names to now be properly loaded

For whatever reason, it was case sensitive and could not find the corresponding template packages. We not hit another error for these tests but it gets much farther now.

* Revert "Stride Particles Tests Updated Visual Tests"

This reverts commit 54e0570601569aa3180dfa5609c08aaf568c43be.

* Revert "Various Graphics Tests Using outdated images [DONT USE]"

This reverts commit dc984b593cea9e8b8752f5da80cfa7b3b89beb89.

* Added skips to GPU rendering related tests that will consecutively fail

For the time being, since these tests are designed for the teamcity agent and that isnt operational, these can be skipped to avoid large amount of failures.

Note: May need feedback on these, might be okay to not gave these skip, regardless of if they always fail.

* Skipped 2 more rendering related test cases, cleaned comments

* Revert "Added skips to GPU rendering related tests that will consecutively fail"

This reverts commit 110a9161ac086c391792e061139d95e04cfc9278.

* Revert "Added skips for graphics related tests to avoid confusion and give additional info for why they fail info"

This reverts commit d598082afe6dd4e9b53ac2102cdc87b8a1b8ece7.

* Removed changed tests back to original tests with more comments

Hopefully can give insight for those seeing issues while SharpFont issues are being looked at

* Revert "Updated template package names to now be properly loaded"

This reverts commit 51c887b0dc0f788ff50f25f1948975af5ae0e166.

* Removed Skips for rendering

* Removed LifetimeNoSimpleConstuctor test and unnecessary constructor

Eideren committed:
Deprecate DataPointer in favor of Span (#2421)

Eideren committed:
feat: TransformComponent.SetWorld, strict equality and fix misc. typos (#2503)

* feat: TransformComponent.SetWorld and fix misc. typos

* Skip unnecessary struct copy

Feralnex committed:
Fixed getting actual bounds and refresh rate of the output (monitor/display) from GraphicsOutput.CurrentDisplayMode when using Direct3D (#2494)

* Fixed getting actual bounds and refresh rate of the output (monitor/display) from GraphicsOutput.CurrentDisplayMode when using Direct3D

* Added exception logging when failed to create Direct3D device in GraphicsOutput.

* Return null if current format doesn't match the requested one

* CurrentDisplayMode initialization refactor

Basewq committed:
[Build] Update Silk.NET from 2.20.0 to 2.22.0 (#2521)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

Eideren committed:
Remove projects still in WIP state from the solution

Eideren committed:
Merge pull request #2131 from Eideren/bepu_integration

feat: Bepu physics integration

Eideren committed:
fix: Add in gizmos on assembly load (#2526)

Jklawreszuk committed:
refactor: FreeImage dependency update (#2531)

* Refactor freeimage files

* Remove unnecessary code

* Remove unused FreeImage wrapper

* fixup! Remove unused FreeImage wrapper

Jklawreszuk committed:
feat: Get LocalAppData path from Environment.SpecialFolder obj (#2511)

VaclavElias committed:
build: Update NuGet.* packages to version 6.12.1 (#2537)

Eideren committed:
fix: Bepu, ensure users require an explicit reference to bepu before being able to use related components and assets (#2534)

Eideren committed:
feat: ExecutionMode control for Flexible Processing (#2535)

Jklawreszuk committed:
feat: Port predicates.c to .NET (#2547)

* Remove unused code

* Fix warnings

* Port predicates.c to .NET

* Add license

* Exclude Stride.Native proj from Stride.Rendering

* fixup! Exclude Stride.Native proj from Stride.Rendering

TranquilAbyss committed:
feat: Made UIElement Key Events public (#2518)

VaclavElias committed:
docs: README.md update - Consistency updates (#2473)

* feat: Readme updates

* feat: Content updates

* feat: Content updated

* feat: Readme updated

* feat: Content updated

* feat: Content improvements

* fix: Some content updates

* feat: Content improvements

* feat: Content updated

* fix: Content updated

* feat: Content updated

* feat: if-building-fails added

* feat: Content updates

* fix: Note word removed

* Update README.md

Co-authored-by: Jakub Ławreszuk <31008367+Jklawreszuk@users.noreply.github.com>

* Update README.md

* Update CODE_OF_CONDUCT.md - Empty line added for better readability.

* Update README.md - Badge added temporarily

* docs: Updates made based on the feedback

---------

Co-authored-by: Jakub Ławreszuk <31008367+Jklawreszuk@users.noreply.github.com>

riibaisu committed:
Replace Xenko logo in MaterialModel.fbx to Stride logo (#1505) (#2546)

Model by CodingSea/MaterialModel@6ea5af27b4d211e699b8b6a45c112d442815d0e4

VaclavElias committed:
chore: Update ServiceWire package to version 5.6.0 (#2548)

- Updated the ServiceWire package version in Directory.Packages.props from 5.5.4 to 5.6.0.

xen2 committed:
[Assets] Fix Stride project upgrader (broken for version 4.0 => 4.2)

Eideren committed:
feat: Process required keyword like MemberRequiredAttribute (#2555)

* feat: Process required keyword like MemberRequiredAttribute

* Add tests

Eideren committed:
fix: very minor prop. grid presentation changes for bepu (#2556)

Jklawreszuk committed:
fix: Remove unused using directive in Stride.Audio (#2557)

Eideren committed:
fix: Bepu, correct minor XML issue (#2558)

Eideren committed:
fix: Remove nuget config workaround for demo project (#2559)

Eideren committed:
fix: CharacterComponent jumping inside trigger boxes (#2562)

Basewq committed:
[Android] Fix invalid configuration in Android solution file. (#2563)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

Jklawreszuk committed:
feat: LockFileEx/UnLockFileEx methods shouldn't be called explicitly (#2564)

TranquilAbyss committed:
fix: Removed Focused loss triggering button Release for keyboard and mouse (#2549)

azeno committed:
[Graphics] Fixes memory leak when creating D3D11 texture with shared options

Nicogo1705 committed:
Allow user to set thread count (#2571)

Eideren committed:
Revert "chore: Update ServiceWire package to version 5.6.0 (#2548)" (#2575)

This reverts commit d9df9dd0be332ef442024349a6586eee069d1b28.

ourabigdev committed:
feat: added outline support for TextBlock (#2582)

* added outline support for TextBlock

added the possibility to add outline to you text block OutlineColor to change the color and OutlineThickness to change Thickness

* Optimize TextBlock outline rendering using nested loops

- No array allocation (Vector2[]) - reduces garbage collection pressure
- More efficient memory usage - uses stack instead of heap

* Remove unrelated whitespace

* Use empty alpha instead of null comparison - color is a struct

---------

Co-authored-by: Eideren <contact@eideren.com>

Doprez committed:
feat: Add numerics type conversions (#2238)

Eideren committed:
docs: Bepu, missing property summaries (#2567)

Eideren committed:
fix: UrlReference triggering quantum change propagation, leading to exceptions with cyclic scene references (#2528)

Jklawreszuk committed:
docs: Update THIRD PARTY.md file (#2577)

Eideren committed:
fix: GridLogViewer empty when messages are appended before the element is on screen (#2525)

Jklawreszuk committed:
feat: Port NuGetResolver.UI to AvaloniaUI (#2512)

* feat: Port NuGetResolver.UI to AvaloniaUI

* fixup! feat: Port NuGetResolver.UI to AvaloniaUI

* fixup! feat: Port NuGetResolver.UI to AvaloniaUI

ClamEater14 committed:
fix: Check repeat key flag before toggling full screen (#2593)

Eideren committed:
fix: bepu, removal of collidables while running a contact event leaves contact tracking in an unexpected state (#2603)

* fix: Re-entry/removal of collidables in contact events

* Finish Flush implementation, add tests

* Update comments and name of SetFreshnessForCurrentActivityStatus

Eideren committed:
Core: ServiceRegistry, GetOrCreate<T> and Remove(instance) (#2599)

Eideren committed:
Core: Refactor service usage following the changes introduced in PR #2599 (#2600)

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2607 from stride3d/all-contributors/add-Nicogo1705

docs: add Nicogo1705 as a contributor for code

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2608 from stride3d/all-contributors/add-ourabigdev

docs: add ourabigdev as a contributor for code

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2610 from stride3d/all-contributors/add-kopffarben

docs: add kopffarben as a contributor for code

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2612 from stride3d/all-contributors/add-Feralnex

docs: add Feralnex as a contributor for code

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2613 from stride3d/all-contributors/add-TranquilAbyss

docs: add TranquilAbyss as a contributor for code

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2614 from stride3d/all-contributors/add-levifmorais

docs: add levifmorais as a contributor for design

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2615 from stride3d/all-contributors/add-tymokvo

docs: add tymokvo as a contributor for doc

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2616 from stride3d/all-contributors/add-Arc-huangjingtong

docs: add Arc-huangjingtong as a contributor for code

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2617 from stride3d/all-contributors/add-minktusk

docs: add minktusk as a contributor for code, and content

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2618 from stride3d/all-contributors/add-timcassell

docs: add timcassell as a contributor for code

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2619 from stride3d/all-contributors/add-dloe

docs: add dloe as a contributor for code

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2620 from stride3d/all-contributors/add-wrshield

docs: add wrshield as a contributor for code

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2621 from stride3d/all-contributors/add-soorMSWE

docs: add soorMSWE as a contributor for code

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2622 from stride3d/all-contributors/add-MechWarrior99

docs: add MechWarrior99 as a contributor for code

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2623 from stride3d/all-contributors/add-kristian15959

docs: add kristian15959 as a contributor for code

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2624 from stride3d/all-contributors/add-YerkoAndrei

docs: add YerkoAndrei as a contributor for code

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2625 from stride3d/all-contributors/add-ComputerSmoke

docs: add ComputerSmoke as a contributor for code

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

VaclavElias committed:
Merge pull request #2626 from stride3d/all-contributors/add-timconner

docs: add timconner as a contributor for code

VaclavElias committed:
Update .all-contributorsrc

VaclavElias committed:
Update README.md

VaclavElias committed:
Merge pull request #2627 from VaclavElias/all-contributors-fydar

docs: Contributor details updated

Jklawreszuk committed:
feat: Crossplatform font loading (#2530)

johang88 committed:
feat: Added support for per frame logical groups in RootEffectRenderFeature (#2597)

VaclavElias committed:
docs: Bepu, XML Comments improvements (#2589)

johang88 committed:
feat: Generalized per view resource binding (#2609)

Basewq committed:
[Build] Fix missing version update of Silk.NET to version 2.22.0 (#2632)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

Eideren committed:
Revert "feat: Crossplatform font loading (#2530)" (#2637)

This reverts commit a06300f5fc261b65ebcf4e9fe1180162462d9cab.

Basewq committed:
fix: Legacy DynamicNavigationMeshSystem bugs and memory leak (#2631)

* [Navigation] Fix detecting DynamicNavigationMeshSystem in NavigationProcessor and fix when DynamicNavigationMeshSystem is disabled

* [Navigation] Ensure native resources are properly disposed from Bullet & Recast

---------

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

Basewq committed:
[Android] Fix registering broadcast receiver on Android 14+ (#2638)

Co-authored-by: Basewq <Basewq@users.noreply.github.com>

net2cn committed:
fix: TextBlock enters an infinite loop if the given available space is too small to fit even one character. (#2598)

Signed-off-by: net2cn <mcopener@gmail.com>

Eideren committed:
fix: Add assets cleanup target when calling dotnet clean, clarify ref out of sync issues (#2629)

Kryptos-FR committed:
refactor: Nullable and Modernization for Stride.Core. & Stride.Core.AssemblyProcessor (#2639)

Eideren committed:
fix: Rendering hang when creating object node (#2640)

Kryptos-FR committed:
refactor: Nullable and Modernization for Core projects (#2643)

* [Core.Mathematics] Modernize code

* [Core.Mathematics] Add tests to Stride.Tests.Simple

* [Core] Cleanup some code

* [Core.CompilerServices] Cleanup some code

* [Core.CompilerServices] Add tests to Stride.Tests.Simple

* [Core.IO] Modernize code

* [Core.MicroThreading] Modernize code

* [Core.Serialization] Modernize code

* [Core.AssemblyProcessor.Tests] Modernize code

* [Core.CompilerServices.Tests] Modernize code

* [Core.Mathematics.Tests] Modernize code

* [Core.Tests] Modernize code

* [Core.Translation] Modernize code

* [Core.Reflection] Modernize code

* Fixes build of Stride.VisualStudio.sln

* [Core.Yaml.Tests] Modernize code

* [Core.Design] Modernize code

* [Core.Design.Tests] Modernize code

* [Core.Mathematics] More annotations

* Fix build of projects that import code from Core.Design

Especially those targeting .NET Framework

Kryptos-FR committed:
Merge branch 'master' into reloads
